packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -195,8 +206,15 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
       endpoint.toBuilder(),
     );
     unawaited(
-      Amplify.asyncConfig.then((_) {
-        _endpointClient.updateEndpoint();
+      Amplify.asyncConfig.then((_) async {
+        final result = await _endpointClient.updateEndpoint();
+        if (result.messageBody.message == 'Accepted') {
```

Comment:
Not needed - if `updateEndpoint` completes, then the request succeeded. It will throw an exception otherwise.

If this fails, though... we have a pretty big problem, right?

Reply:
Indeed, Analytics wouldn't be working properly.  I updated the code to catch and throw the exception. 

```
unawaited(
      Amplify.asyncConfig.then((_) async {
        try {
          await _endpointClient.updateEndpoint();
          _hubEventController.add(
            AnalyticsHubEvent.pinpointEndpointIdSet(
              AnalyticsHubEventPayload(pinpointEndpointId: fixedEndpointId),
            ),
          );
        } on Exception catch (e) {
          throw AnalyticsException(
            'Error updating Pinpoint Endpoint',
            underlyingException: e,
          );
        }
      }),
```